lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1604,7 +1604,7 @@ public function get_collection_params() {
 			),
 		);
 		$params['status'] = array(
-			'default'           => 'publish',
+			'default'           => 'attachment' === $this->post_type ? 'inherit' : 'publish',
```

Comment:
> Does it just remove the attachments I can't view (thus I get 6 results rather than the default 10)

Yes, which is a consistent pattern we have throughout the controllers.